design-proposals/mass_machine_type_transition.md

```diff
@@ -0,0 +1,302 @@
+$ rpm -qa | grep qemu-kvm-core
+qemu-kvm-core-8.0.0-5.el9.x86_64
+$ /usr/libexec/qemu-kvm -machine ?
+Supported machines are:
+pc                   RHEL 7.6.0 PC (i440FX + PIIX, 1996) (alias of pc-i440fx-rhel7.6.0)
+pc-i440fx-rhel7.6.0  RHEL 7.6.0 PC (i440FX + PIIX, 1996) (default) (deprecated)
+q35                  RHEL-9.2.0 PC (Q35 + ICH9, 2009) (alias of pc-q35-rhel9.2.0)
+pc-q35-rhel9.2.0     RHEL-9.2.0 PC (Q35 + ICH9, 2009)
+pc-q35-rhel9.0.0     RHEL-9.0.0 PC (Q35 + ICH9, 2009)
+pc-q35-rhel8.6.0     RHEL-8.6.0 PC (Q35 + ICH9, 2009) (deprecated)
+pc-q35-rhel8.5.0     RHEL-8.5.0 PC (Q35 + ICH9, 2009) (deprecated)
+pc-q35-rhel8.4.0     RHEL-8.4.0 PC (Q35 + ICH9, 2009) (deprecated)
+pc-q35-rhel8.3.0     RHEL-8.3.0 PC (Q35 + ICH9, 2009) (deprecated)
+pc-q35-rhel8.2.0     RHEL-8.2.0 PC (Q35 + ICH9, 2009) (deprecated)
+pc-q35-rhel8.1.0     RHEL-8.1.0 PC (Q35 + ICH9, 2009) (deprecated)
+pc-q35-rhel8.0.0     RHEL-8.0.0 PC (Q35 + ICH9, 2009) (deprecated)
+pc-q35-rhel7.6.0     RHEL-7.6.0 PC (Q35 + ICH9, 2009) (deprecated)
+none                 empty machine
+```
+This means each workload must be updated to use a `pc-q35-rhel9.x.x` machine type.
+
+## Goals
+* Create an opt-in workflow that automates mass converting machine types of VMs
+	* if currently running, update VM machine type and update its Status to alert
+  the user the VM must be restarted for the change to take effect
+	* alternatively, if currently running, take VM offline immediately and update
+  machine type
+	* if not currently running, update machine type immediately
+* Allow user to specify certain VMs for updating machine type automation
+(e.g. by namespace)
+
+## Non Goals
+While these additions are beneficial to cluster-admins who wish to use the mass
+machine type transition (MMTT), these will not be included in the initial
+implementation:
+* Allowing user to filter VMs by running state: a cluster admin may want to only
+update VMs that are offline.
+* Subcommands that allow the user to monitor and manage the status of the MMTT
+job and affected VMs
+
+These will eventually be implemented in follow ups once the initial design is implemented.
+
+Additionally, this design proposal does not cover the method in which users
+(cluster-admins) will be informed that they have VMs with unsupported machine
+types.
+
+## User Stories
+As a cluster-admin, I want an automated way to update the machine type of all
+VMs with unsupported machine types without interrupting my workflow.
+As a cluster-admin, I want to be able to include only certain VMs to be updated
+(e.g. by namespace, label-selector).
+As a cluster-admin, I want to be in control of the behavior of running VMs being
+updated; I can determine whether I want to manually restart running VMs being
+updated or let the automation restart all of them for me. 
+
+## Repos
+[kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+Create a new virtctl command `virtctl update machine-types` that will allow
```

Comment:
Are we planning on adding more update-able fields in the future? If not, this could probably be a single command, like `update-machine-types`.
Edit: see comment in future phases for a better idea

Reply:
Originally it was a single command, but some earlier discussion brought up a two layered approach which involved the separate `update machine-types